S7.Net/PLC.cs

```diff
@@ -229,28 +366,33 @@ public ErrorCode Open()
                         return ErrorCode.WrongCPU_Type;
                 }
 
+                //COTP Setup
                 _mSocket.Send(bSend1, 22, SocketFlags.None);
-                if (_mSocket.Receive(bReceive, 22, SocketFlags.None) != 22)
+                var response = ReadCOTP_TPDU();
```

Comment:
Shouldn't check for 22 before calling ReadCOTP_TPDU ?

Reply:
Nope. I'm pretty sure I removed this check but it came back somehow. I'll fix 😊